roles/lib_dyn/library/dyn_record.py

```diff
@@ -233,27 +258,48 @@ def main():
     # All states will need a list of the exiting records for the zone.
     dyn_node_records = get_any_records(module, dyn_node)
 
+    # get_record_values()
+    #dyn_values = get_record_values(dyn_node_records)
+    dyn_values = introspect_records(dyn_node_records)
+
     if module.params['state'] == 'list':
-        module.exit_json(changed=False,
-                         records=get_record_values(
-                             dyn_node_records,
-                         ))
+        module.exit_json(changed=False, dyn_records=dyn_values)
 
-    if module.params['state'] == 'present':
+    elif module.params['state'] == 'absent':
+        # If there are any records present we'll want to delete the node.
+        if dyn_node_records:
+            dyn_node.delete()
+            # Publish the zone since we've modified it.
+            dyn_zone.publish()
+            module.exit_json(changed=True, msg="Removed node %s from zone %s" % (dyn_node_name, module.params['zone']))
+
+        module.exit_json(changed=False)
+
+    elif module.params['state'] == 'present':
+
+        # configure the TTL variable:
+        # if use_zone_ttl, use the default TTL of the account.
+        # if TTL == None, don't check it, set it as 0 (api default)
+        # if TTL > 0, ensure this TTL is set
+        if module.params['use_zone_ttl']: #or isinstance(module.params['record_ttl'], str):
+            user_param_ttl = dyn_zone.ttl
+        elif not module.params['record_ttl']:
+            user_param_ttl = 0
```

Comment:
This will cause the same always updating case that you're trying to work around. The value of 0 means set the value of the node ttl to be the same as the default for the zone. Its much simpler to leave the default as 0 and simply compare the `dyn_zone.ttl` to the node's ttl when `module.params['record_ttl'] == 0`. The `module.params['use_zone_ttl']` isn't necessary.